hardware/openwebnet/bt_openwebnet.cpp

```diff
@@ -846,3 +847,1439 @@ bool bt_openwebnet::IsWriteFrame(){return (frame_type == WRITE_FRAME);}
 bool bt_openwebnet::IsPwdFrame(){return (frame_type == PWD_FRAME);}
 bool bt_openwebnet::IsOKFrame(){return (frame_type == OK_FRAME);}
 bool bt_openwebnet::IsKOFrame(){return (frame_type == KO_FRAME);}
+
+bool bt_openwebnet::ParseData(char* data, int length, vector<bt_openwebnet>& messages)
+{
+	string buffer = string(data, length);
+	size_t begin = 0;
+	size_t end = string::npos;
+	do {
+		end = buffer.find(OPENWEBNET_END_FRAME, begin);
+		if (end != string::npos) {
+			bt_openwebnet message(buffer.substr(begin, end - begin + 2));
+			messages.push_back(message);
+			begin = end + 2;
+		}
+	} while (end != string::npos);
+
+	return true;
+}
+
+string bt_openwebnet::getWhoDescription(string& who)
+{
+	if (who == "0") {
+		return "Scenario";
+	}
+	else if (who == "1") {
+		return "Lighting";
+	}
+	else if (who == "2") {
+		return "Automation";
+	}
+	else if (who == "3") {
+		return "Load control";
+	}
+	else if (who == "4") {
+		return "Temperature control";
+	}
+	else if (who == "5") {
+		return "Burglar alarm";
+	}
+	else if (who == "6") {
+		return "Door entry system";//the Who 6 addresses directly to cameras instead the Who 7 was written for frames sent to the web server
+	}
+	else if (who == "7") {
+		return "Video door entry system";
+	}
+	else if (who == "9") {
+		return "Auxiliary";//not documented
+	}
+	else if (who == "13") {
+		return "Gateway interfaces management";
+	}
+	else if (who == "14") {
+		return "Light shutter actuator lock";
+	}
+	else if (who == "15") {
+		return "Scenario Scheduler Switch";
+	}
+	else if (who == "16") {
+		return "Audio";
+	}
+	else if (who == "17") {
+		return "Scenario programming";
+	}
+	else if (who == "18") {
+		return "Energy management";
+	}
+	else if (who == "22") {
+		return "Sound diffusion";
+	}
+	else if (who == "24") {
+		return "Lighting management";
+	}
+	else if (who == "25") {
+		return "Dry contact/IR Detection";
+	}
+	else if (who == "1000") {
+		return "Diagnostic";//not documented
+	}
+	else if (who == "1001") {
+		return "Automation diagnostic";//not documented
+	}
+	else if (who == "1004") {
+		return "Thermoregulation diagnostic";// documented with who=4
+	}
+	else if (who == "1013") {
+		return "Device diagnostic";//not documented
+	}
+	else if (who == "1018") {
+		return "Energy device diagnostic";//not documented
+	}
+
+	return "Unknown who : " + who;
+}
+
+
+string bt_openwebnet::getWhatDescription(string& who, string& what, vector<string>& whatParameters)
```

Comment:
Could you make the variables const where possible? Like 'std::string &who'
This is also for all functions ;)